book/src/spec.md

```diff
@@ -10,33 +10,33 @@ We use Halo2/IPA with [Pasta curves](https://github.com/zcash/pasta) developed b
 Let `C(x; w) ⟶ 0/1` be a circuit with up to `m` gates. The circuit is represented as polynomials over the chosen curve's **scalar field**, following [plonk-ish arithmetization](https://zcash.github.io/halo2/concepts/arithmetization.html). Commitments are generated over the curve's **base field**.
 
 ### 1.2 Elliptic curves
-||Name|Scalar field| Base field|Purpose|Instantiation|
-|-|-|-|-|-|-|
-|$E_I$|Inner curve|$\mathbb{F}_q$|$\mathbb{F}_p$|ECC gadget| [Pallas](https://github.com/zcash/pasta#pallasvesta-supporting-evidence)
-|$E_M$|Main curve|$\mathbb{F}_p$|$\mathbb{F}_q$|Action and VP circuits| Vesta|
-|$E_O$|Outer curve|$\mathbb{F}_q$|$\mathbb{F}_p$|Accumulation circuit| Pallas|
+|Name|Base field| Scalar field|Purpose|Instantiation|
+|-|-|-|-|-|
+|$E_p$|$\mathbb{F}_p$|$\mathbb{F}_q$|ECC gadget, Accumulation circuit| [Pallas](https://github.com/zcash/pasta#pallasvesta-supporting-evidence)
+|$E_q$|$\mathbb{F}_q$|$\mathbb{F}_p$|Action and VP circuits| Vesta|
 
 ### 1.3 Proving system interfaces
 ||Interface|Description|
 |-|-|-|
-|__Preprocess__|`preproc(C) ⟶ desc_C`|`C` is turned into a *circuit description*, which is all data the verifier needs to verify a proof. It includes the verifier key `C_vk`, but not only that.|
-|__Prove__|`P(C, x, w) ⟶ π`||
-|__Verify__|`V(desc_C, x, π) ⟶ 0/1`||
+|__Generate Verifier key__|`keygen_vk(C) ⟶ vk`|`C` is turned into a *circuit description* or *verifying key* `vk`, a succint representation of the circuit that the verifier uses to verify a proof|
+|__Generate Proving key__|`keygen_pk(C, vk) ⟶ pk`|Generate a proving key from a verifying key and an instance of circuit|
+|__Prove__|`P(C, pk, x, w) ⟶ π`|Prove that a circuit is satisfied given instance `x` and witness `w`|
+|__Verify__|`V(vk, x, π) ⟶ 0/1`|Verify the proof|
 
 ## 2. Notes
-Note is an immutable particle of the application state.
+Note is an immutable particle of the application state in the UTXO model.
 
 ### 2.1 Note structure
-|Variable|Type/size|Description|
+|Variable|Type|Description|
 |-|-|-|
-|`value_base`||Value base represents the note type|
-|`app_data_dynamic`||Commitment to the note's extra data|
-|`v`|${0..2^{64} - 1}$|The quantity of fungible value|
-|`cm_nk`||Commitment to the nullifier key that will be used to derive the note's nullifier|
-|`ρ`|$\mathbb{F}_p$|An old nullifier from the same Action description (see Orchard)|
-|`ψ`|$\mathbb{F}_p$|$ψ = PRF^{ψ}(0, rseed, ρ)$|
-|`is_merkle_checked`|bool|Dummy note flag. It indicates whether the note's commitment Merkle path should be checked when spending the note.|
-|`rcm_note`|${0..2^{255} - 1}$|A random commitment trapdoor $rcm\_note = PRF^{\texttt{rcm\_note}}(1, rseed, ρ)$|
+|`note_type`|$E_p$|Identifier of the note's application|
+|`app_data_dynamic`|$\mathbb{F}_p$|Encoding of the application's extra data|
+|`v`|u64|The quantity of fungible value specific to a note type|
```

Comment:
I'm not sure the value is specific to a note type. The difference between 1 eth and 1 btc is not in the value part (1) but in the type part (btc/eth)

Reply:
Also using `u64` (as in rust type) adds inconsistency as all other types are defined in a mathy way (either by how they are interpreted or what they are). I got the range from the Orchard specification

Reply:
The _fungibility_ is specific to a note type

Reply:
Hm, maybe we could say it like that? "The quantity of fungible value. The fungibility is specific to a note type"

Reply:
I find "the quantity of fungible value" really confusing. I updated the definition

Reply:
I can't see the update. Could you link the commit?

Reply:
https://github.com/anoma/taiga/pull/189/commits/3eaee3a28dc21b3beac4829614e154ff01554e99